src/evolution.jl

```diff
@@ -67,124 +131,68 @@ function evolve(s, τ, N, B, L, N_sites, Δx, Δm², p, x, Δp, theta_nu, ψ, sh
             end
         end
-        
-        # if shape_name!=="none" #specfic to the inhomogenous case Test4
-        #     # compute the avg expectation value of Sz at all sites
-        #     sz_tot = expect(ψ, "Sz")
-        #     sz = mean(sz_tot)
-        # else 
-            # compute expectation value of Sz (inbuilt operator in ITensors library) at the first site on the chain
-            # sz = expect(ψ, "Sz"; sites=1)
-        # end 
-
-        if shape_name!=="none" #specfic to the inhomogenous case Test4
-            # compute the avg expectation value of Sz at all sites
-            sz_tot = expect(ψ, "Sz")  # Compute Sz for each site and store the values in sz_tot
-            half_N = div(N_sites, 2)  # Calculate half of the number of sites
-            sz = mean(sz_tot[1:half_N])  # Take the mean of the first half of the sz_tot array
-        
-        else 
-            # compute expectation value of Sz (inbuilt operator in ITensors library) at the first site on the chain
-            sz = expect(ψ, "Sz"; sites=1)
-        end 
-
-        # compute expectation value of sy and sx using S+ and S- (inbuilt operator in ITensors library) at the first site on the chain
-        if p == zeros(N_sites, 3) #for rogerro's case only (b/c S+ S- needed to keep conservation of QN number)
-            sy = -0.5 *im * (expect(complex(ψ), "S+"; sites=1) - expect(complex(ψ), "S-"; sites=1)) #re-check
-            sx = 0.5 * (expect(ψ, "S+"; sites=1) + expect(ψ, "S-"; sites=1)) #recheck
-        else 
-            sy = expect(complex(ψ), "Sy"; sites=1)
-            sx = expect(ψ, "Sx"; sites=1)
-        end
 
-        # add an element sz to ... 
-        push!(Sz_array, sz) # .. the end of Sz array 
-        push!(Sy_array, sy) # .. the end of Sy array 
-        push!(Sx_array, sx) # .. the end of Sx array 
-        
-        # survival probability for a (we took first) neutrino to be found in its initial flavor state (in this case a spin down)
-        prob_surv = 0.5 * (1 - 2 * sz)
+        # compute the avg expectation value of Sz at all sites
+        sz_tot = expect(ψ, "Sz")  # Compute Sz for each site and store the values in sz_tot
 
-        # add an element prob_surv to the end of  prob_surv_array 
-        push!(prob_surv_array, prob_surv)
+        # compute expectation value of sy and sx (inbuilt operator in ITensors library) at all sites on the chain
+        sy_tot = expect(complex(ψ), "Sy")
+        sx_tot = expect(ψ, "Sx")
 
-        if B[1] == 1
-            println("$t $sz")
-        else 
-            println("$t $prob_surv")
-        end
+        push!(Sz_array, sz_tot)  # Add all elements of sz_tot to Sz_array
+        push!(Sy_array, sy_tot)  # Add all elements of sz_tot to Sz_array
+        push!(Sx_array, sx_tot)  # Add all elements of sz_tot to Sz_array
         
+        #survival probability for all sites (neutrino) to be found in its initial flavor state
+        prob_surv_tot = 0.5 * (1 .- 2 .* sz_tot)
```

Comment:
This definition of survival probability only applies to specific initial conditions. I would remove these lines and relegate survival probability to post-processing. I think rho_ee would probably be a good diagnostic to print instead.